src/libstd/fs.rs

```diff
@@ -674,6 +716,41 @@ impl DirEntry {
     /// The exact text, of course, depends on what files you have in `.`.
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn path(&self) -> PathBuf { self.0.path() }
+
+    /// Return the metadata for the file that this entry points at.
+    ///
+    /// This function may not require an extra system call on some platforms
+    /// but other platforms will need to perform a system call for this.
+    ///
+    /// This function will not traverse symlinks if this entry points at a
+    /// symlink.
+    #[unstable(feature = "dir_entry_ext", reason = "recently added API")]
+    pub fn metadata(&self) -> io::Result<Metadata> {
+        self.0.metadata().map(Metadata)
+    }
+
+    /// Return the file type for the file that this entry points at.
+    ///
+    /// This function may not require an extra system call on some platforms
+    /// but other platforms will need to perform a system call for this.
+    ///
+    /// This function will not traverse symlinks if this entry points at a
+    /// symlink.
+    #[unstable(feature = "dir_entry_ext", reason = "recently added API")]
+    pub fn file_type(&self) -> io::Result<FileType> {
+        self.0.file_type().map(FileType)
+    }
+
+    /// Returns the bare file name of this directory entry without any other
+    /// leading path component.
+    #[unstable(feature = "dir_entry_ext", reason = "recently added API")]
+    pub fn file_name(&self) -> OsString {
```

Comment:
Like I pointed out in the RFC (a little lately, sorry for that). If the entry is a directory or something else, is `file_name` is still a good method name ?